lib/include/tree_sitter/api.h

```diff
@@ -79,6 +79,15 @@ typedef struct TSInput {
   TSInputEncoding encoding;
 } TSInput;
 
+typedef struct TSParseState {
+  void *payload;
+} TSParseState;
```

Comment:
One *suggestion* in the linked issue was knowing the current status of errors detected, but I feel that that's a bit hard to nail down properly since the parser *could* in theory recover with an error-free tree in the event an error was detected. This was more to set us up with a struct that can be amended later on, so that the signature of these callbacks does not change, and it retains ABI compatibility

Reply:
Ok, that use case does make sense. But what would get passed back in `ParseState`? I would think that you'd want to control that in the caller: like `ParseOptions::new().error_callback(my_callback)`.

Reply:
Yeah, that makes a lot more sense, I can't really think of any stateful information that should be passed into the interrupt callback then, so I'll remove these wrapper structs